### PR TITLE
HOCS-3952 remove the use of the Junit4 and Retry libraries from the project

### PR DESCRIPTION
**Pull request by nickmcmahon01**
This commit removes the use of the Junit4 and Retry libraries from the project.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -49,11 +48,13 @@ dependencies {
...
 	testImplementation('com.github.tomakehurst:wiremock-standalone:2.18.0')
 }
 
 jar {
 	enabled = false
 }
+
+test {
+	useJUnitPlatform()
+}
```
Comment (LiamMacP) [inline build.gradle:60]: New line please...

File: src/main/java/uk/gov/digital/ho/hocs/templates/application/RestHelper.java [hunk 2]
```diff
@@ -35,14 +33,12 @@ public RestHelper(RestTemplate restTemplate, @Value("${hocs.basicauth}") String
...
     }
 
-    @Retryable(maxAttemptsExpression = "${retry.maxAttempts}", backoff = @Backoff(delayExpression = "${retry.delay}"))
     public <R> R get(String serviceBaseURL, String url, Class<R> responseType) {
         log.info("RestHelper making GET request to {}{}", serviceBaseURL, url, value(LogEvent.EVENT, REST_HELPER_GET));
         ResponseEntity<R> response = restTemplate.exchange(String.format("%s%s", serviceBaseURL, url), HttpMethod.GET, new HttpEntity<>(null, createAuthHeaders()), responseType);
         return response.getBody();
     }
 
-    @Retryable(maxAttemptsExpression = "${retry.maxAttempts}", backoff = @Backoff(delayExpression = "${retry.delay}"))
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/templates/application/RestHelper.java:45]: Can you remove `retry.maxAttempts` and `retry.delay` from the app props?

Comment (LiamMacP) [reply src/main/java/uk/gov/digital/ho/hocs/templates/application/RestHelper.java:45]: Is there a library that can be pulled now?

File: src/test/java/uk/gov/digital/ho/hocs/templates/client/documentclient/DocumentClientTest.java [hunk 3]
```diff
@@ -42,10 +43,10 @@ public void shouldGetTemplate() {
         verifyNoMoreInteractions(restHelper);
     }
 
-    @Test(expected = HttpClientErrorException.NotFound.class)
+    @Test
     public void shouldThrowNotFoundExceptionWhenGetTemplate() {
         doThrow(HttpClientErrorException.NotFound.class).when(restHelper).get(eq(serviceBaseURL), eq(String.format("/document/%s/file", uuid)), eq(ByteArrayResource.class));
-        documentClient.getTemplate(uuid);
+        assertThrows(HttpClientErrorException.NotFound.class, () -> documentClient.getTemplate(uuid));
```
Comment (LiamMacP) [inline src/test/java/uk/gov/digital/ho/hocs/templates/client/documentclient/DocumentClientTest.java:49]: Sweet!

**Review by LiamMacP — CHANGES_REQUESTED** [review]
Please seperate the one commit into 2 so they can be rebased on merge.

File: src/main/java/uk/gov/digital/ho/hocs/templates/application/RestHelper.java (hunk 2 above)
Comment (nickmcmahon01) [reply src/main/java/uk/gov/digital/ho/hocs/templates/application/RestHelper.java]: no, they aren't in there?! maybe passed in from the deployment?!!!

**Pull request closed without merging**